share/osutil/file_linux.go

```diff
@@ -81,6 +81,74 @@ func extractProcRootPath(pid int, input string, inTest bool) (string, error) {
 	}
 }
 
+// The retryResolveSymlink splits the path into parts and checks each segment for symlinks, updating the path as it resolves them.
+func retryResolveSymlink(procRoot, prefix, remaining string) (bool, string) {
```

Comment:
Can this logic support this case?
- Resolving `/a/b/c` in a container.
- `/a` is a symlink pointing to `/d/e/f`
- `/d/e` is another symlink pointing to `/g/h/i`

The expected result should be `/g/h/i/f/b/c`.  The flow is:

1. `/a/b/c` => `/d/e/f` + `/b/c`
2. `/d/e/f/b/c` => `/g/h/i` + `/f/b/c` => `/g/h/i/f/b/c`

But from your code, it might become `/d/e/f/b/c`?

1. `/a` => `/d/e/f`
2. `/d/e/f` + `/b` => `/d/e/f/b`
3. `/d/e/f/b` + `/c` => `/d/e/f/b/c`

```
# ls -al a 
lrwxrwxrwx 1 root root 6 Jan 25 16:22 a -> /d/e/f
# ls -al /d/e
lrwxrwxrwx 1 root root 6 Jan 25 16:22 /d/e -> /g/h/i
# stat /a/b/c
Device: 20002eh/2097198d	Inode: 58720946    Links: 1
# stat /g/h/i/f/b/c
Device: 20002eh/2097198d	Inode: 58720946    Links: 1
```

Reply:
Hi @holyspectral ,
Thanks for the review!
You're correct, the correct way is to recursive back to re-run it from beginning.
In new commit, I add a traverse count to limit the number of traverse, and guarantee retry from the beginning.